org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/models/DeviceWatchdogModel.java

```diff
@@ -1,219 +1,219 @@
+	 * be a scannable name. The expression will be evaluated using JEXL which is
+	 * available through the IExpressionService. Any expression
+	 */
+	private String expression; // e.g. 'beamcurrent >= 1.0 && !portshutter.equalsIgnoreCase("Closed")'
+	private String message;
+	
+	// t
+	private String countdownName; // e.g. "topup", "countdown" PV likely to be SR-CS-FILL-01:COUNTDOWN which is in s
+	
+	// c in ms
+	private long   cooloff;       // time in ms before topup for which the scan should be paused.
+	
+	// w in ms
+	private long   warmup;        // time in ms after topup the scan should wait before starting.
+	
+	// p in ms
+	private long   period = 10*60*1000; // Period in ms, default is 10min
+	
+	// Tf in ms
+	private long   topupTime = 15*1000; // The time that a topup takes. This is varible but in normal mode <= 15s
+	
+	// The name of the mode pv, if any. 
+	private String modeName;            // If this is set the PV will be checked to ensure that the topup mode is as expected.
+	
+	public String getCountdownName() {
+		return countdownName;
+	}
+	public void setCountdownName(String monitorName) {
+		this.countdownName = monitorName;
+	}
+	public long getCooloff() {
+		return cooloff;
+	}
+	public void setCooloff(long cooloff) {
+		this.cooloff = cooloff;
+	}
+	public long getWarmup() {
+		return warmup;
+	}
+	public void setWarmup(long warmup) {
+		this.warmup = warmup;
+	}
+	public String getMessage() {
+		return message;
+	}
+	public void setMessage(String message) {
+		this.message = message;
+	}
+	public String getExpression() {
+		return expression;
+	}
+	public void setExpression(String expression) {
+		this.expression = expression;
+	}
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + (int) (cooloff ^ (cooloff >>> 32));
+		result = prime * result + ((countdownName == null) ? 0 : countdownName.hashCode());
+		result = prime * result + ((expression == null) ? 0 : expression.hashCode());
+		result = prime * result + ((message == null) ? 0 : message.hashCode());
+		result = prime * result + (int) (warmup ^ (warmup >>> 32));
+		return result;
+	}
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		DeviceWatchdogModel other = (DeviceWatchdogModel) obj;
+		if (cooloff != other.cooloff)
+			return false;
+		if (countdownName == null) {
+			if (other.countdownName != null)
+				return false;
+		} else if (!countdownName.equals(other.countdownName))
+			return false;
+		if (expression == null) {
+			if (other.expression != null)
+				return false;
+		} else if (!expression.equals(other.expression))
+			return false;
+		if (message == null) {
+			if (other.message != null)
+				return false;
+		} else if (!message.equals(other.message))
+			return false;
+		if (warmup != other.warmup)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)